### ISSUE TITLE
Incosistent newline-per-chained-call rule 

### ISSUE THREAD

**Issue by xamoom-pali** [eslint/eslint#9138]
* ESLint Version: v4.4.1
* Node Version: v6.0.0
* npm Version: v3.8.6

**What parser (default, Babel-ESLint, etc.) are you using?**
eslint / ember-cli-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'module'
  },
  extends: [
    'airbnb-base',
    'plugin:ember/recommended'
  ],
  env: {
    browser: true,
  },
  rules: {
    "ember/named-functions-in-promises": "off",
    "ember/no-observers": "off"
  },
  settings: {

    'import/core-modules': [
      './config/environment',
      'ember',
      'ember-load-initializers',
      'ember-resolver',
      'ember-data/model',
      'ember-data/attr',
      'ember-data/relationships',
      'ember-data/serializers/json-api',
      'ember-data/-private/transforms/string',
      'ember-cp-validations',
      'ember-cp-validations/validators/base',
      'ember-cp-validations/validators/format',
      'ember-simple-auth/mixins/authenticated-route-mixin',
      'ember-simple-auth/mixins/data-adapter-mixin',
      'ember-simple-auth/authenticators/oauth2-password-grant',
      'ember-simple-auth/authorizers/base',
      'ember-moment/computeds/moment',
      'ember-moment/computeds/format',
      'ember-moment/computeds/locale',
      'model-fragments',
      'model-fragments/attributes'
    ],
  },
};

```
</details>

**What did you do? Please include the actual source code causing the issue.**
I am getting ESlint errors in a somehow inconsistent way.
There are 2 functions. One of them is analysed as correct, while the second one is marked
as problematic. 

See the ESlint complaint below.

```js

     // this function is not correct according to ESlint
    _doSave() {
      this.indicateProcessStart();

      this._updateRemovedLocalizedInfos()
        .then(() => {
          return this._updateDirtyLocalizedInfos();
        }).then(() => {
          return this._updateSystem();
        }).then((system) => {
          set(this, 'initialLocalizedSystemInfosLength', get(system, 'localizedSystemInfos.length'));
          this.send('success', get(this, 'i18n').t('components.alert.success').toString());
        }).catch((error) => { // <---- complaint here
          this.send('error', error);
        }).finally(() => { // <---- complaint here
          this.indicateProcessCompletion();
        });
    },

    // this function is correct according to ESlint while using .catch() in a very same way. Why?
    _updateDirtyLocalizedInfos() {
      return new Ember.RSVP.Promise((resolve) => {
        const dirtyLocalizedSystemInfos = A();
        ...

        const updateNextLSI = function () {
          if (isEmpty(dirtyLocalizedSystemInfos)) {
            resolve();
            return;
          }

          const lsiToBeUpdated = dirtyLocalizedSystemInfos.objectAt(0);

          lsiToBeUpdated.save()
            .then(() => {
              dirtyLocalizedSystemInfos.removeObject(lsiToBeUpdated);
              updateNextLSI();
            }).catch((e) => {  // <---- ESlint not complaining here. Why?
              dirtyLocalizedSystemInfos.removeObject(lsiToBeUpdated);
              updateNextLSI();
            });
        };

        updateNextLSI();
      });
    },

```

**What did you expect to happen?**
I expect ESlint either complain in both cases or not complain at all.

**What actually happened? Please include the actual, raw output from ESLint.**
```
[eslint] Expected line break before `.catch`. (newline-per-chained-call)
```
Am I missing anything here or is it a bug with ESlint rule itself? Thank you

**Comment by not-an-aardvark** [comment]
This is occurring because `eslint-config-airbnb-base` uses the [`ignoreChainWithDepth: 4`](https://github.com/airbnb/javascript/blob/68cdc593620f2cdc3b5a0e9a7b960bc519db9eb9/packages/eslint-config-airbnb-base/rules/style.js#L208) option, so only chains of "length" longer than 4 should be reported. Your first chain is longer than your second chain, and your second chain has a length of less than 4, so only the first chain is reported.

**Comment by xamoom-pali** [comment]
Thank you very much! Now it all makes sense. Are we closing the thread now?
